cpp/oneapi/dal/test/engine/linalg.hpp

```diff
@@ -20,3 +20,4 @@
 #include "oneapi/dal/test/engine/linalg/loops.hpp"
 #include "oneapi/dal/test/engine/linalg/umath.hpp"
 #include "oneapi/dal/test/engine/linalg/dot.hpp"
+#include "oneapi/dal/test/engine/linalg/random.hpp"
```

Comment:
Having random generators inside `linalg` confuses me. Is it a common-used practice?

Reply:
It's randomly generated matrices with specific properties

Reply:
Matrices can allow not only linear algebra operations. Seems like you put random matrix creation here because the matrix is also defined here. I suggest separating this module on `engine/matrix`, `engine/linalg`, and `engine/random`.

Reply:
Agree in general, but I'd like to stick here to numpy conventions: having data container in the root package, but linalg and random operations in submodules, such design has already proven its convenience. That's what I suggest:
1. Move `test/engine/linalg/matrix.hpp` -> `test/engine/matrix.hpp`, `test::engine::linalg::matrix` -> `test::engine::matrix`
2. Introduce `test/engine/random/*.hpp` and `test/engine/random.hpp`, move all RNG functions there
3. Remove extra namespaces like `linalg` or `random`, but keep separation on directories
    - I don't believe we need them for testing purpose, but it requires people to write `namespace la = ..`, `namespace rng = ...` in every test.

I don't want to do it as part of this PR, as it affect most of the tests. I'll create another one to refactor test engine once lapack is merged. There are other things I wanted to do, e.g., move `test/engine/dataframe_*.hpp` files to separate directory. It's good moment to do that.